docs/source/poet/sgx_primer.rst

```diff
@@ -0,0 +1,208 @@
+identity of the enclave. The enclave measurement can also be referred to as the
+"Enclave Identity" or "MRENCLAVE".
+
+Sealing Authority
+-----------------
+
+“The Sealing Authority is an entity that signs the enclave prior to
+distribution, typically the enclave builder.” [2]_ Upon creation of an enclave,
+the enclave is presented with a signed certificate containing an Enclave
+Measurement and the public key of the Sealing Authority. The enclave verifies
+the signature for the certificate and then compares the “proposed” measurement
+in the certificate against the measurement it computed locally through creation.
+If they are the same, the enclave stores the public key as the Sealing Authority
+for the enclave. The sealing authority can also be refered to as the "Sealing
+Identity" or "MRSIGNER".
+
+Execution
+---------
+
+Execution of enclave code follows the process below:
+
+1. The untrusted application code calls an untrusted proxy function defined by
+   the enclave API.
+#. The proxy function prepares input arguments for the enclave.
+#. The proxy function makes an ECall and transfers control to the trusted
+   enclave code.
+#. The trusted enclave code executes, making OCalls as needed to interact with
+   OS and other untrusted code.
+#. Trusted enclave code prepares output data for untrusted application code.
+#. Trusted enclave code transfers control back to untrusted application code.
+
+Attestation
+===========
+
+Attestation
+  The process of demonstrating that a piece of software has been properly
+  instantiated on the platform.
+
+Within SGX, attestation is, “the mechanism by which another party can gain
+confidence that the correct software is running within an enclave on an enabled
+platform.” [2]_
+
+Enclave Reports
+---------------
+
+Enclaves can generate a data structure referred to as a "report" which can be
+used for attestation. When creating a report, the measurement of the target
+enclave (the intended receiver of the report) is required. Enclave reports
+contain:
+
+* The enclave’s identity (Sealing Authority and Enclave Measurement)
+* Additional enclave attributes
+* Additional developer-defined information
+* A message authentication code (MAC)
+
+Intra-Platform Attestation
+--------------------------
+
+Intra-Platform attestation refers to attestation between two enclaves on the
+same platform. That is, they are both running on the same Intel® SGX enabled
+processor and host.
+
+1. A report is created for a given target enclave on the same platform.
+2. The target enclave verifies the report.
+3. A secure channel is then constructed between the two enclaves using standard
+   encryption methods.
+
+Inter-Platform Attestation
+--------------------------
+
+Inter-Platform attestation refers to attestation between two enclaves running on
+different platforms, most likely connected by a network connection. This form of
+attestation is used by PoET and Intel® provides an implementation of this service
```

Comment:
add a comma here: "attestation is used by PoET**,** and Intel® provides an implementation", right after "PoET", to clear up semantics